cylc/uiserver/data_store_mgr.py

```diff
@@ -140,26 +144,33 @@ def update_workflow_data(self, topic, delta, w_id):
                 loop_cnt += 1
                 continue
         if topic == 'shutdown':
+            self.delta_store_to_queues(w_id, topic, delta)
             self.workflows_mgr.stopping.add(w_id)
-            self.w_subs[w_id].stop()
             return
-        delta_time = getattr(
-            delta, 'time', getattr(delta, 'last_updated', 0.0))
-        # If the workflow has reloaded recreate the data
-        # otherwise apply the delta if it's newer than the previously applied.
-        if delta.reloaded:
-            if topic == WORKFLOW:
-                self.data[w_id][topic].CopyFrom(delta)
-            else:
-                self.data[w_id][topic] = {
-                    ele.id: ele
-                    for ele in delta.deltas
-                }
-            self.data[w_id]['delta_times'][topic] = delta_time
-        elif delta_time >= self.data[w_id]['delta_times'][topic]:
-            apply_delta(topic, delta, self.data[w_id])
-            self.data[w_id]['delta_times'][topic] = delta_time
-            self.reconcile_update(topic, delta, w_id)
+        for field, sub_delta in delta.ListFields():
+            delta_time = getattr(sub_delta, 'time', 0.0)
+            # If the workflow has reloaded clear the data before
+            # delta application.
+            if sub_delta.reloaded:
+                if field.name == WORKFLOW:
+                    self.data[w_id][field.name].Clear()
+                else:
+                    self.data[w_id][field.name].clear()
```

Comment:
What's going on here, is there a difference between `Clear` and `clear`?

Reply:
Ones a Protobuf element (i.e. `PbWorkflow`), so uses `Clear`, the other's a dictionary.